src/search.cpp

```diff
@@ -350,16 +351,19 @@ void Thread::search() {
 
   multiPV = std::min(multiPV, rootMoves.size());
 
-  int hIdx = (idx - 1) % 20; // helper index, cycle after 20 threads
-
   // Iterative deepening loop until requested to stop or the target depth is reached
   while (   (rootDepth += ONE_PLY) < DEPTH_MAX
          && !Signals.stop
          && (!Limits.depth || Threads.main()->rootDepth / ONE_PLY <= Limits.depth))
   {
-      // skip half of the plies in blocks depending on game ply and helper index.
-      if (idx && ((rootDepth / ONE_PLY + rootPos.game_ply() + phase[hIdx]) / skipsize[hIdx]) % 2)
-          continue;
+      // Distribute search depths across the threads, picking the next depth to search
+      // for each thread in a way to make each thread searching on a different depth.
```

Comment:
@mcostalba I find the sentence 'to make each thread searching on a different depth.' a bit misleading: it is not true that each thread will searching on a different (root)-depth, in fact often a part of them will do search with same rootDepth.  The principle is actually that each singular rootDepth is searched by exactly the half of helper-threads while the other half will skip it.